assets/js/background.js

```diff
@@ -198,17 +310,24 @@ chrome.runtime.onInstalled.addListener(function(details) {
   }
 });
 
-// listen for tabs url changes
+/**
+ * Listen for tabs url changes
+ */
 chrome.tabs.onUpdated.addListener(handleOnUpdated);
 
 /**
  * Message handler for inter instance messaging
  */
-var messageHandler = function messageHandler(request, sender, sendResponse) {
-  // Reload contexts on successful login message from login page
-  if(request.type === 'anyfetch::loginSuccessful') {
-    refreshTabs();
-    sendResponse();
+chrome.runtime.onMessage.addListener(function messageHandler(request, sender, sendResponse) {
+  console.log(arguments);
+  var handlers = {
+    'anyfetch::backgroundFindContext': findContext,
+    'anyfetch::backgroundGetResults': getResults,
+    'anyfetch::backgroundToggleContextItem': toggleContextItem,
+    'anyfetch::backgroundLoginSuccessful': loginSuccessful
+  };
+  if(request.type && handlers[request.type]) {
+    return handlers[request.type](request, sender, sendResponse);
   }
-};
-chrome.runtime.onMessage.addListener(messageHandler);
+  return false;
```

Comment:
Useless I think, but it was probably meant to explicitly return a falsy value to Chrome (the async response thing)